pypdf/generic/_base.py

```diff
@@ -344,7 +344,11 @@ def __new__(
         cls, value: Union[str, Any] = "0.0", context: Optional[Any] = None
     ) -> "FloatObject":
         try:
-            value = float(str_(value))
+            if isinstance(value, bytes):
+                value = value.decode()
+            else:
+                value = str_(value)
+            value = float(value)
             return float.__new__(cls, value)
         except Exception as e:
             # If this isn't a valid decimal (happens in malformed PDFs)
```

Comment:
I guess this has to be replaced by "float"?

Reply:
Normally part of the PR on float.